### DISCUSSION TITLE
Missing tiles at 180-epsilon East

### DISCUSSION THREAD

**Discussion by mbe57** [mapsforge/mapsforge#1293] [Q&A] [answered]
Hi Emux,
just observed that the Java version for PC omits a number of tiles (x=2^^n-1), while the same Android version seems to miss nothing. What is the technical difference here? And how can the Java/PC version be fixed?
Test: OAM Chukotka, Wrangel Island south coast is the easiest spot to see the deviation.
TXs and cheer
Michael

**Comment by mbe57** [comment]
PS: can be reproduced with Mapsforge own maps, but they are HUGE.
Well, best is Sea East of Australia from OAM, only88 MB. Fiji tiles are tilt.

**Comment by devemux86** [comment, marked as answer]
Related to #643, #780.

It appears in Mapsforge (not [VTM](https://github.com/mapsforge/vtm)) on all platforms.

I added an [option](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-core/src/main/java/org/mapsforge/core/util/Parameters.java) for coordinates validation (#1294).

---

It will be available in next [Cruiser](https://wiki.openstreetmap.org/wiki/Cruiser).

**Comment by mbe57** [comment]
Great you know the root cause on the spot!
Funny enough on my S10, Android 12, there is NO issue. On two other devices Cruiser has the issue.
I do hope that the fix will also be available in the MapsForge Java libraries (for MOBAC, e.g.).
Can you pls. advise?
TXs and cheers
Michael

**Reply by devemux86** [reply]
> Can you pls. advise?

They can use the new parameter via the snapshot builds.
See the [integration guide](https://github.com/mapsforge/mapsforge/blob/master/docs/Integration.md).

**Comment by mbe57** [comment]
PS: on the S10 I am using the Cruiser-GL - does this use the VTM then? This would explain ,,,

**Reply by devemux86** [reply]
- Cruiser uses Mapsforge
- Cruiser GL uses VTM

**Comment by mbe57** [comment]
TXs a lot, Emux!

**Comment by mbe57** [comment]
Hello Emux,
Robert took the snapshot and recompiled MOBAC , but the failed tiles persist. Yet, we've got error messages:

2022-02-16 17:22:58,819 WARN  org.mapsforge.map.layer.renderer.DatabaseRenderer Exception: invalid longitude: 180.00000034375
2022-02-16 17:22:58,820 ERROR mobac.mapsources.MapsforgeMapSource Failed to render image 16383/3800/z14

Clearly a numerical problem that - somehow - is resolved in VTM, getting the same 180+ longitudes, while MapsForge does not fix them.
Can you pls. advise?
TXs and cheers
Michael

**Reply by devemux86** [reply]
See my post above: https://github.com/mapsforge/mapsforge/discussions/1293#discussioncomment-2180012

> I added an [option](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-core/src/main/java/org/mapsforge/core/util/Parameters.java) for coordinates validation (https://github.com/mapsforge/mapsforge/pull/1294).

There is a new `Parameters.VALIDATE_COORDINATES` option that can be set to `false` to disable the coordinates validation.
(see the [changelog](https://github.com/mapsforge/mapsforge/blob/master/docs/Changelog.md))

**Comment by mbe57** [comment]
Sorry for my slkow comprehension, Emux. Now I got it :-)
TXs again.
Cheers
Michael
